File: longest_common_prefix/code.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
        StringBuilder res = new StringBuilder();

        Arrays.sort(strs);

        char [] s1 = strs[0].toCharArray();
        char [] s2 = strs[strs.length-1].toCharArray();

        for(int i=0; i<s1.length; ++i){
            if(s1[i] != s2[i]){
                break;
            }
            res.append(s1[i]);
        }
        return res.toString();

    }
}

File: sqrt(x)/code.cpp
class Solution {
public:
    int mySqrt(int x) {
        long long l = 0;
        long long r = x;
        while(r-l>1){
            long long m = (l+r)/2;
            if(m*m == x){
                return (int)m;
            } else if(m*m < x){
                l = m;
            } else {
                r = m;
            }
        }
        if(r*r == x){
            return (int)r;
        }
        return (int)l;
    }
};
